Source/WebCore/page/EventHandler.cpp

```diff
@@ -5090,19 +5089,16 @@ HandleUserInputEventResult EventHandler::handleTouchEvent(const PlatformTouchEve
             *pointerTarget, event, index, !index, *document->windowProxy(), { 0, 0 });
 #endif
 
-        if (frame.ptr() != targetFrame) {
-            // pagePoint should always be relative to the target elements containing frame.
-            pagePoint = documentPointForWindowPoint(*targetFrame, point.pos());
-        }
+        // pagePoint should always be relative to the target elements containing frame.
+        pagePoint = LayoutPoint(targetFrame->view()->windowToContents(point.pos()));
 
         float scaleFactor = targetFrame->pageZoomFactor() * targetFrame->frameScaleFactor();
 
-        int adjustedPageX = lroundf(pagePoint.x() / scaleFactor);
-        int adjustedPageY = lroundf(pagePoint.y() / scaleFactor);
+        FloatPoint adjustedPagePoint = FloatPoint(pagePoint.x() / scaleFactor, pagePoint.y() / scaleFactor);
```

Comment:
LayoutPoint has an `operator FloatPoint()`, so I think this should just work.

```suggestion
        FloatPoint adjustedPagePoint { pagePoint };
        adjustedPagePoint.scale(1 / scaleFactor);
        
```